E2E: revise the WordPress.com Free signup spec to the new, simplified signup flow, take 2.

worldomonation opened the pull request:
This is a re-attempt of https://github.com/Automattic/wp-calypso/pull/73618. Title and description have been reused.

Related to paYE8P-2bD-p2

## Proposed Changes

This PR repurposes the existing `Plans: Free` flow to to the newly simplified `Free Signup` flow.

This new flow is sufficiently different from existing ones that new coverage was deemed necessary in p1676573321667029-slack-CQD1HH4MA.

Key changes:
- addition of `FreeSignupFlow` POM, similar to the existing `StartSiteFlow`.
- added coverage of the `Free Signup` flow available at `setup/free/freeSetup`.

## Testing Instructions

Ensure the following build configurations are passing:
  - [ ] Pre-Release E2E

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

worldomonation committed:
packages/calypso-e2e/src/lib/flows/
- add new POM to represent the Free Signup flow.

worldomonation committed:
test/e2e/specs/plans/plans__signup-free.ts
- move the test file to `specs/onboarding`.

test/e2e/specs/onboarding/signup__free.ts
- rework the spec to cover the new simplified free signup flow.

worldomonation committed:
Add link to project thread

worldomonation committed:
Update the method name to `enterSiteName`.

worldomonation committed:
packages/calypso-e2e/src/lib/flows/free-signup-flow.ts
- do not specify exact match

worldomonation committed:
packages/calypso-e2e/src/lib/flows/free-signup-flow.ts
- fix selector

dpasque commented on test/e2e/specs/onboarding/signup__free.ts:
```diff
@@ -0,0 +1,118 @@
+
+		it( 'Land in Launchpad', async function () {
+			// Processing steps often takes longer than the default timeout.
+			await page.waitForURL( /launchpad/, { timeout: 20 * 1000 } );
```
**Suggestion (non-blocking)**: In rare occasions, while doing this flow myself, I've had this take a pretty long time. Any risk here in bumping this to 30 seconds just to be really safe? Since this is pre-release, my thought is we should avoid false positives as much as possible!

worldomonation replied:
Done, fixed in 5a19b39a963f34c2d38b3b41f5f5e1c7170e6d24.

dpasque commented on test/e2e/specs/onboarding/signup__free.ts:
```diff
@@ -0,0 +1,118 @@
+		let sidebarComponent: SidebarComponent;
+
+		it( 'User has the selected free domain', async function () {
+			const urlRegex = `${ testUser.siteName }.wordpress.com`;
```
**Nit (non-blocking):** This is technically just a string, not a regex (which is fine, because the `toMatch` function handles substring searches). [Ref](https://jestjs.io/docs/expect#tomatchregexp--string)

Can we update the variable name to something like `expectedDomain`?

worldomonation replied:
I renamed it to `expectedURL` in 5a19b39a963f34c2d38b3b41f5f5e1c7170e6d24. 

Also renamed the test step because the user didn't explicitly select a domain name per sé.

dpasque reviewed (approved):
Beautifully done! Simple, effective, and reads like a charm! 💛 

Left two non-blockers, otherwise GTG!

worldomonation committed:
test/e2e/specs/onboarding/signup__free.ts
- extend timeout for launchpad
- rename test step
- rename variable to `expectedURL`
